.NET/Microsoft.Recognizers.Text.Choice/Models/ChoiceModel.cs

```diff
@@ -19,17 +20,25 @@ protected ChoiceModel(IParser parser, IExtractor extractor)
 
         public List<ModelResult> Parse(string query)
         {
-            var extractResults = Extractor.Extract(query);
-            var parseResults = extractResults.Select(r => Parser.Parse(r));
+            try
+            {
+                var extractResults = Extractor.Extract(query);
+                var parseResults = extractResults.Select(r => Parser.Parse(r));
 
-            return parseResults.Select(pr => new ModelResult()
+                // o can be null, add this part to try.
+                return parseResults.Select(pr => new ModelResult()
+                {
+                    Start = pr.Start.Value,
+                    End = pr.Start.Value + pr.Length.Value - 1,
+                    Resolution = GetResolution(pr),
+                    Text = pr.Text,
+                    TypeName = ModelTypeName,
+                }).ToList();
+            }
+            catch (Exception)
             {
-                Start = pr.Start.Value,
-                End = pr.Start.Value + pr.Length.Value - 1,
-                Resolution = GetResolution(pr),
-                Text = pr.Text,
-                TypeName = ModelTypeName,
-            }).ToList();
+                return null;
```

Comment:
Code should never return null. If no extracted entity, we should return an empty list.